### PR TITLE
feat(*): update the design and add a new masthead component to all pages

### PR DESCRIPTION
**Pull request by chrisvogt**
This PR introduces a lot of design updates. If any of these changes are controversial I can separate them into separate PRs to discuss and work on. This PR...

- [x] Updates the document `<title>` content so the page name always reads first.
- [x] Rearranges the document head.
- [x] Removes Raleway in preference of the default Materialize font (Roboto).
- [x] Updates all headlines to be GD Sherpa.
- [x] Refactors styles to use a new GoDaddy theme.
- [x] Adds a new, custom banner for the Open Graph (Twitter/Facebook) share previews.
- [x] Removes unnecesary Open Graph tags (these values are already inferred).
- [x] Removes the "jobs" page altogether in preference of the official Careers site.
- [x] Updates the primary navigation links to use relative paths, opposed to absolute paths.
- [x] Adds a new `masthead` component to all layouts. It contains the `<h1>` headlines.
- [x] Updates all layouts to use the same header and footer templates.
- [x] Adds the subhead "Joining forces with the open source community".
- [x] Polishes the home page copy.
- [x] Fixes HTML5 validation errors around `<ul>` nesting in the Code of Conduct page.

#### Open Graph Banner

OG Banner

#### Desktop

_Home_

home

_Projects_

projects

_Blog Index_

blog-index

_Blog post_

blog-post

#### Mobile

_Home, Samsung_

screen shot 2018-05-04 at 3 39 07 am

_Code of Conduct, Pixel 2_

screen shot 2018-05-04 at 3 39 59 am

_Blog Index, iPhone X_

screen shot 2018-05-04 at 3 38 16 am

_Blog Post, iPhone X_

screen shot 2018-05-04 at 3 38 30 am

### CODE REVIEW

File: _includes/block/head.html [hunk 1]
```diff
@@ -1,43 +1,33 @@
...
 
-  <!-- Compiled and minified CSS -->
-  <link href="https://fonts.googleapis.com/css?family=Raleway:400,400i,800,900" rel="stylesheet">
-  <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">  
-  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/materialize/0.100.2/css/materialize.min.css">
+  <title>{% if page.title %}{{ page.title }} — {% endif %}{{ site.name }}</title>
+  <meta name="description" content="{{ site.description }}">
 
-  <link rel="stylesheet" href="{{ '/assets/css/main.css' | relative_url }}" />
+  <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
```
Comment (mayank23) [inline _includes/block/head.html]: should we leverage `preload` for these assets? (speed up first time visits)

**Comment by chrisvogt** [comment]
I'm not very familiar with that. Reading about it now [on MDN](
https://developer.mozilla.org/en-US/docs/Web/HTML/Preloading_content). What
do you suggest?

File: _includes/block/head.html (hunk 1 above)
Comment (tarkatronic) [reply _includes/block/head.html]: I would personally lean toward no, just to err on the side of caution. According to the MDN page @chrisvogt linked to, it's still in Candidate Recommendation status, so only being experimentally implemented now, with no support in Internet Explorer yet. And according to [the W3C document](https://w3c.github.io/preload/):
> This is a work in progress and may change without any notices.
> 
> Implementers SHOULD be aware that this document is not stable.

Of course, on the other hand, if it doesn't break older browsers without support, then I suppose it won't hurt anything. 👍

**Comment by mayank23** [comment]
Hey @chrisvogt 😄  , actually I'm thinking that in order to keep this PR confined, we can have a separate PR for resource prioritization (preload, etc). That sounds fine?

File: _includes/block/head.html (hunk 1 above)
Comment (mayank23) [reply _includes/block/head.html]: hey @tarkatronic  😄 , the basic definition already exists in the spec https://html.spec.whatwg.org/multipage/links.html#link-type-preload. Further details for preload is in Candidate Recommendation status.

**Pull request merged**